test/async-hooks/test-improper-order.js

```diff
@@ -51,8 +51,12 @@ if (process.argv[2] === 'child') {
   child.stderr.on('data', (d) => { errData = Buffer.concat([ errData, d ]); });
   child.stdout.on('data', (d) => { outData = Buffer.concat([ outData, d ]); });
 
-  child.on('close', common.mustCall((code) => {
-    assert.strictEqual(code, 1);
+  child.on('close', common.mustCall((code, signal) => {
+    if (signal) {
+      console.log(`Child closed with signal: ${signal}`);
```

Comment:
Do we know why the signal can be truthy? And if so, what signal is it? Normally this should not happen unless the child process get killed, but this test does not actively kill the child process, so it seems to be something outside the test killing the process. So technically one can break the test by making the child process abort (SIGABRT) and this test would not be able to catch the change of behavior any more.

Reply:
sounds good, can you send it as a commit.

Reply:
FYI @nodejs/platform-aix

Reply:
There was some parallel discussion in https://github.com/nodejs/node/pull/58478#discussion_r2132170010, so I have opened https://github.com/nodejs/node/pull/58601